git_aggregator/config.py

```diff
@@ -141,9 +143,20 @@ def load_config(config, expand_env=False, force=False):
     conf = kaptan.Kaptan(handler=kaptan.HANDLER_EXT.get(file_extension))
 
     if expand_env:
+        environment = {}
+        if env_file is not None and os.path.isfile(env_file):
+            with open(env_file) as env_file_handler:
+                for line in env_file_handler:
+                    line = line.strip()
+                    if line.startswith('#'):
+                        continue
+                    if '=' in line:
+                        key, value = line.split('=')
+                        environment.update({key.strip(): value.strip()})
+        environment.update(os.environ)
```

Comment:
Shouldn't the variables in the file have priority over the environment?
ie: variable is defined in both places, I'd expect the one in the file to be used

Reply:
From my practice it's beter to have the environment variables with higher priority. Let say that the environment is a more "local" thing than a config file. It's like that that it works with docker-compose for exemple. It allows to have a shared env file with common configuration and local adjustments through environment.

Reply:
AFAIK in docker-compose the environment variables are not passed unless explictly say so (using shell `-e VARIABLE`).
And IMO it's the fact that it has to be explicit that makes it the more "local" thing.

But here, all the env vars are substituted, and the file is more explicit than the actual environment, so IMO it has to have priority

Reply:
No, in docker-compose you don't have to do anything else than setting the variable in the environment:

``` console
export VARIABLE=andrei
```

Then VARIABLE will be taken from the environment even if it's in the .env file. 

A full exemple of how docker-compose works with varaibles:

```console
[andrei@stealth test]$ cat docker-compose.yml 
version: '3'
services:
  db:
    image: ${DB_SERVER_IMAGE}
[andrei@stealth test]$ cat .env 
DB_SERVER_IMAGE=postgres:10.0
[andrei@stealth test]$ docker-compose config
services:
  db:
    image: postgres:10.0
version: '3.0'

[andrei@stealth test]$ export DB_SERVER_IMAGE=postgres:12.0
[andrei@stealth test]$ docker-compose config
services:
  db:
    image: postgres:12.0
version: '3.0'

```

When I say more "local" I mean that the environment is really a local configuration - it's not shared with others. On he other hand the env file may be shared with others.

Imagine the situation where you have a common configuration in e .env file containing 20 variables and you want to customize for you only one of those variables. If the variables don't contain sensible information then you'll want to keep the .env file under version control. In this situation editing it locally is not the best option. On the other way you may tweak the one variable in your environment. 

To be honest for me both ways are good, I'm just trying to find the more standard and logical way to do it.

Reply:
Thanks for the clarification.

I was thinking in the env that goes into the container, not the docker-compose substitution.
I see your point, I agree 👍